src/components/contracts/utils.js

```diff
@@ -6,11 +6,36 @@ export const formatSpeed = speed => {
   return Number(speed) / 10 ** 12;
 };
 
-export const formatTimestamp = timestamp => {
+export const formatTimestamp = (timestamp, timer, state) => {
   if (+timestamp === 0) {
     return '';
   }
-  return moment.unix(timestamp).format('L');
+  if (state !== CONTRACT_STATE.Running) {
+    return '';
+  }
+  const startDate = moment.unix(timestamp).format('L');
+  const { days, hours, minutes } = timer;
+  if (timer.isRunning && (days || hours || minutes)) {
```

Comment:
do you need `(days || hours || minutes)`?
```suggestion
  if (timer.isRunning) {
```

Reply:
yes, at the first render for contracts that are already expired `isRunning === true`, but days, hours, minutes === 0. So it takes a few seconds until isRunning changes to false. To avoid such behaviour I added this check

Reply:
so maybe drop isRunning then, making days/hours/minutes more reliable source of truth?

And please don't forget about seconds

Reply:
sounds reasonable